File: app/src/main/java/uz/mayasoft/marta/wallet/pin2/security/IPFPinCodeHelper.java
package uz.mayasoft.marta.wallet.pin2.security;

import android.content.Context;

import uz.mayasoft.marta.wallet.pin2.security.callbacks.PFPinCodeHelperCallback;


public interface IPFPinCodeHelper {

    void encodePin(Context context, String pin, PFPinCodeHelperCallback<String> callBack);

    void checkPin(Context context, String encodedPin, String pin, PFPinCodeHelperCallback<Boolean> callback);


    void delete(PFPinCodeHelperCallback<Boolean> callback);

    void isPinCodeEncryptionKeyExist(PFPinCodeHelperCallback<Boolean> callback);

}

File: app/src/main/java/com/example/marta/ui/pin/security/IPFPinCodeHelper.kt
package com.example.marta.ui.pin.security

import android.content.Context
import com.example.marta.ui.pin.security.calbacs.PFPinCodeHelperCallback


interface IPFPinCodeHelper {

    fun encodePin(
        context: Context?,
        pin: String?,
        callBack: PFPinCodeHelperCallback<String>
    )

    fun checkPin(
        context: Context?,
        encodedPin: String?,
        pin: String?,
        callback: PFPinCodeHelperCallback<PFResult<Boolean>>
    )

    fun delete(callback: PFPinCodeHelperCallback<PFResult<Boolean>>)

    fun isPinCodeEncryptionKeyExist(callback: PFPinCodeHelperCallback<PFResult<Boolean>>)
}